### PR TITLE
metrics: add telemetry to track mcd_host_os_and_version

### PR DESCRIPTION
**Pull request by kikisdeliveryservice**
Allowing us to track the host OS and version will help in troubleshooting
specific version errors and track errors across versions.

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
I tried running: `MAKE=make make generate-in-docker` which seemed to be going ok but then threw an error:
```
make: *** No rule to make target 'generate'.  Stop.
make: *** [Makefile:81: generate-in-docker] Error 2
```
and if i git status i have a bunch of deleted files???:
```
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

    deleted:    manifests/benchmark/clusterRoleBindingPrometheusOperator.yaml
    deleted:    manifests/benchmark/deploymentPrometheusOperator.yaml
    deleted:    manifests/benchmark/prometheusPrometheus.yaml
    deleted:    manifests/benchmark/roleBindingConfigPrometheus.yaml
    deleted:    manifests/benchmark/roleBindingSpecificNamespacesPrometheus.yaml
    deleted:    manifests/benchmark/roleConfigPrometheus.yaml
    deleted:    manifests/benchmark/roleSpecificNamespacesPrometheus.yaml
```

I don't understand what's happening, could someone please help?

**Comment by LiliC** [comment]
@kikisdeliveryservice 
You need to run `make clean` first and then `make generate-in-docker`. Let me know if this works, I tried it locally and works for me. We should also update the README, seems to be very vague right now.

File: jsonnet/telemeter/metrics.jsonnet [hunk 1]
```diff
@@ -92,6 +92,9 @@
   // subscription_sync_total is the number of times an OLM operator
   // Subscription has been synced, labelled by name and installed csv
   '{__name__="subscription_sync_total"}',
+   // mcd_host_os_and_version is the OS of the node that the MCD is running on for ex: RHCOS, CentOS,
+   // unsupported.  If the OS is RHCOS, the RHCOS version is also returned
+  '{__name__="mcd_host_os_and_version"}',
```
Comment (brancz) [inline jsonnet/telemeter/metrics.jsonnet]: Will this produce a time-series per node?

**Comment by kikisdeliveryservice** [comment]
@brancz here is the metric

Screenshot from 2019-10-31 13-13-48

**Comment by kikisdeliveryservice** [comment]
I can't get the Makefile work. @runcom is going to try

**Comment by runcom** [comment]
So, using Docker (on osx) yields the same as @LiliC:
```
➜  telemeter git:(pr/origin/257) ✗ git st
On branch pr/origin/257
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   docs/data-collection.md
	modified:   docs/sample-metrics.md
	modified:   manifests/benchmark/statefulSetTelemeterServer.yaml
	modified:   manifests/client/deployment.yaml

no changes added to commit (use "git add" and/or "git commit -a")
```

@kikisdeliveryservice are you using podman on fedora? any chance you'd install docker from upstream and retry? (Dan Walsh is gonna kill me but 🤷‍♂ )

**Comment by kikisdeliveryservice** [comment]
ok finally!  squashed and ready to go :)
thanks @runcom :smile:

**Comment by brancz** [comment]
This metric seems like it’s on a per node basis. That’s too high of a cardinality, we will need to roll that up into aggregates.

**Comment by kikisdeliveryservice** [comment]
So I guess when we do metrics for the MCC, we could report this on a pool basis and report back to telemetry.  If I'm understanding correctly the whole point is that we dont want continuous metrics from each node all the time.  The user still has access to this via Prom.

Will close and reopen if necessary.

cc: @runcom 

/close

**Pull request closed without merging**